apps/rush-lib/src/cli/logic/PackageChangeAnalyzer.ts

```diff
@@ -9,30 +9,22 @@ import {
 } from '@microsoft/package-deps-hash';
 import { Path } from '@microsoft/node-core-library';
 
-import RushConfiguration from '../../data/RushConfiguration';
 import { RushConstants } from '../../RushConstants';
+import RushConfiguration from '../../data/RushConfiguration';
 
 export class PackageChangeAnalyzer {
   public static _instance: PackageChangeAnalyzer | undefined;
 
   // Allow this function to be overwritten during unit tests
   public static getPackageDeps: (path: string, ignoredFiles: string[]) => IPackageDeps;
-  public static rushConfig: RushConfiguration;
 
   private _data: Map<string, IPackageDeps>;
 
-  public static get instance(): PackageChangeAnalyzer {
-    if (!PackageChangeAnalyzer._instance) {
-      PackageChangeAnalyzer._instance = new PackageChangeAnalyzer();
-    }
-    return PackageChangeAnalyzer._instance;
-  }
-
   public static reset(): void {
     PackageChangeAnalyzer._instance = undefined;
   }
 
-  public constructor() {
+  public constructor(private _rushConfiguration: RushConfiguration) {
```

Comment:
>private [](start = 21, length = 7)

This TypeScript feature is a bad practice.  We should have a tslint rule to forbid it.

OOP is about factoring your program state into objects, and then associating operations with the right objects (versus loose static functions).  Some famous guy had a quote like "If the data structures are designed right, then the meaning of the methods will be obvious."  In this view, a class's fields should be grouped together at the top, and managed carefully.  Hiding them inside a constructor parameter (or property getter in C#) works against that philosophy.